test/parsers.js

```diff
@@ -544,139 +544,82 @@ test('#parseError()', function (t) {
-  })
-
-  t.test('error library frame 1 line', function (t) {
-    var err = new Error()
-    validateParseCallsite(t, err, {
-      isApp: true,
-      isError: false,
-      lines: 1,
-      pre: [],
-      line: '    var err = new Error()',
-      post: []
-    })
-  })
-
-  /**
-   * TRACE LIBRARY FRAME
-   */
-  t.test('error library frame 0 lines', function (t) {
-    var err = new Error()
-    validateParseCallsite(t, err, {isApp: false, isError: false, lines: 0})
-  })
-
-  t.test('error library frame 1 line', function (t) {
-    var err = new Error()
-    validateParseCallsite(t, err, {
-      isApp: false,
-      isError: false,
-      lines: 1,
-      pre: [],
-      line: '    var err = new Error()',
-      post: []
+  var cases = [
+    {isApp: true, isError: true, lines: 0},
+    {isApp: true, isError: true, lines: 1},
+    {isApp: true, isError: true, lines: 2},
+    {isApp: true, isError: true, lines: 3},
+    {isApp: true, isError: true, lines: 4},
+    {isApp: true, isError: true, lines: 5},
+    {isApp: true, isError: false, lines: 0},
+    {isApp: true, isError: false, lines: 1},
+    {isApp: true, isError: false, lines: 2},
+    {isApp: true, isError: false, lines: 3},
+    {isApp: true, isError: false, lines: 4},
+    {isApp: true, isError: false, lines: 5},
+    {isApp: false, isError: true, lines: 0},
+    {isApp: false, isError: true, lines: 1},
+    {isApp: false, isError: true, lines: 2},
+    {isApp: false, isError: true, lines: 3},
+    {isApp: false, isError: true, lines: 4},
+    {isApp: false, isError: true, lines: 5},
+    {isApp: false, isError: false, lines: 0},
+    {isApp: false, isError: false, lines: 1},
+    {isApp: false, isError: false, lines: 2},
+    {isApp: false, isError: false, lines: 3},
+    {isApp: false, isError: false, lines: 4},
+    {isApp: false, isError: false, lines: 5}
+  ]
+
+  cases.forEach(function (opts) {
+    t.test('#parseCallsite() ' + JSON.stringify(opts), function (t) {
+      validateParseCallsite(t, opts)
     })
   })
 })
 
-function validateParseCallsite (t, err, opts) {
+function validateParseCallsite (t, opts) {
+  // before 2
+  // before 1
+  var err = new Error()
+  // after 1
+  // after 2
+
+  switch (opts.lines) {
+    case 1:
+      opts.pre = []
+      opts.line = '  var err = new Error()'
+      opts.post = []
+      break
+    case 2:
+      opts.pre = ['  // before 1']
+      opts.line = '  var err = new Error()'
+      opts.post = []
+      break
+    case 3:
+      opts.pre = ['  // before 1']
+      opts.line = '  var err = new Error()'
+      opts.post = ['  // after 1']
+      break
+    case 4:
+      opts.pre = ['  // before 2', '  // before 1']
+      opts.line = '  var err = new Error()'
+      opts.post = ['  // after 1']
+      break
+    case 5:
+      opts.pre = ['  // before 2', '  // before 1']
+      opts.line = '  var err = new Error()'
+      opts.post = ['  // after 1', '  // after 2']
+      break
+  }
```

Comment:
An alternative to doing this here would be a case factory like this:

```js
function makeCase (data) {
  var lines = data.lines
  if (lines) {
    data.line = '  var err = new Error()'
    var pre = data.pre = []
    var post = data.post = []
    var before = Math.ceil((lines - 1) / 2)
    var after = Math.floor((lines - 1) / 2)
    for (var i = before; i > 0; i--) {
      pre.push('  // before ' + i)
    }
    for (var i = 1; i <= after; i++) {
      post.push('  // after ' + i)
    }
  }
  return data
}
```

Reply:
I considered that, but I also wanted the tests to contain as little magic as possible (I already feel that I'm pushing the limit in this case). The more complex this logic becomes the bigger the risk that I introduce an error that in worse case gives us a false positive test.